cmds/find/find.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Find finds files. It is similar to the Unix command. It uses REs, not globs,
+// for matching.
+//
+// OPTIONS:
+//     -d: enable debugging in the find package
+//     -mode integer-arg: match against mode, e.g. -mode 0755
+//     -type: match against a file type, e.g. -type f will match files
+//     -name: name is an RE, e.g. -name '.*[a-z][0-9]$' will match files ending in a letter and number.
```

Comment:
I think this would be too finicky and surprising considering that a lot of filenames contain a '.' and that '.' has special meaning in regex. Consider this find:

    find --name ha.a

Would also match "haha".

Or, if you consider finding files which begin with a period:

    find --name .*
    find --name ..*
    find --name \..*

None of these would work. In the last one, most shells would convert '\.' to '.' before sending as argv. These two would work:

    find --name \\..*
    find --name '\..*'

I believe findutils has it right. Use the `-regex` flag for regular expressions and the `-name` flag for non-regex matching. You can still decide not to support globs in `-name`.

Having both commands is trival to add with this function: https://godoc.org/regexp#QuoteMeta

Reply:
The more I think about, I'm leaning towards agreeing with @rjoleary. -name should be glob based, i.e. `filepath.Match(pattern, name string)`, and regex should be a separate CLI arg.

Reply:
I'm not a fan of all these switches. I'm happy to stick with Glob until someone demonstrates a use case for REs.

Let's be careful.

"The key to problem-solving on the UNIX system is to identify the right primitive operations and to
put them at the right place. UNIX programs tend to solve general problems rather than special cases. In a
very loose sense, the programs are orthogonal, spanning the space of jobs to be done (although with a fair
amount of overlap for reasons of history, convenience or efficiency). Functions are placed where they will
do the most good: there shouldn’t be a pager in every program that produces output any more than there
should be filename pattern matching in every program that uses filenames.
One thing that UNIX does not need is more features. It is successful in part because it has a small
number of good ideas that work well together. Merely adding features does not make it easier for users to
do things it just makes the manual thicker. The right solution in the right place is always more effective
than haphazard hacking."

http://harmful.cat-v.org/cat-v/unix_prog_design.pdf

I realize some of this is obsolete, but still.

Reply:
And, yep, I'm using this with globs now and it bites. But I'll go with it.